crates/polars-core/src/chunked_array/logical/categorical/merge.rs

```diff
@@ -113,7 +113,7 @@ impl GlobalRevMapMerger {
 fn merge_local_rhs_categorical<'a>(
     categories: &'a Utf8Array<i64>,
     ca_right: &'a CategoricalChunked,
-) -> Result<(impl Iterator<Item = Option<u32>> + 'a, Arc<RevMapping>), PolarsError> {
```

Comment:
We always collected the iterator, so it is faster to use `apply` as this will optimally traverse the chunks and iterate per chunk.

@c-peters FYI.